### PR TITLE
Support history -> alert tags

### PR DESCRIPTION
**Pull request by psiinon**

### CODE REVIEW

**Comment by psiinon** [comment]
Yeah, looking at that now..

File: zap/src/main/java/org/zaproxy/zap/extension/alert/ExtensionAlert.java [hunk 1]
```diff
@@ -206,7 +208,22 @@ public void alertFound(Alert alert, HistoryReference ref) {
...
+
+            int srcHrefId = alert.getSourceHistoryId();
+            if (srcHrefId > 0) {
+                // Copy any tags that are intended for alerts
+                List<String> tags = HistoryReference.getTags(srcHrefId);
+                Map<String, String> ctags = new HashMap<>(alert.getTags());
+                for (String tag : tags) {
+                    if (tag.startsWith(ALERT_TAG_PREFIX) && tag.contains("=")) {
+                        String tagValue = tag.substring(ALERT_TAG_PREFIX.length());
+                        String[] kv = tagValue.split("=");
```
[the remark this answers is not in the document]
Comment (psiinon) [reply zap/src/main/java/org/zaproxy/zap/extension/alert/ExtensionAlert.java:221]: I guess the characters after the 2nd `=` are ignored. That ok or would we like them to be preserved in the value?

[the remark this answers is not in the document]
Comment (kingthorin) [reply zap/src/main/java/org/zaproxy/zap/extension/alert/ExtensionAlert.java:221]: Doesn't matter I guess. If it's truncated that should probably be mentioned in the help.

[the remark this answers is not in the document]
Comment (kingthorin) [reply zap/src/main/java/org/zaproxy/zap/extension/alert/ExtensionAlert.java:221]: Should probably also be covered in the unit test

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]
Thanks I think this handling of multiple equal signs is better.

**Pull request merged**